java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -279,7 +280,7 @@ class DatabaseImpl(
                     ON fields.is_collection = 1
                     AND collection_entries.collection_id = field_values.value_id
                 LEFT JOIN number_primitive_values
-                    ON fields.type_id = 1 AND number_primitive_values.id = field_value_id
+                    ON (fields.type_id = 1 OR fields.type_id = 6) AND number_primitive_values.id = field_value_id
```

Comment:
note additional check here

Reply:
I'd recommend this instead: `ON fileds.type_id IN $NUMBER_TYPE_IDS`

where `NUMBER_TYPE_IDS = "(1, 6)"` (or even something built up from the actual enum values instead)

Reply:
Nice! Done.